Framework/MDAlgorithms/src/CompactMD.cpp

```diff
@@ -0,0 +1,140 @@
+#include "MantidMDAlgorithms/CompactMD.h"
+#include "MantidAPI/IMDIterator.h"
+using namespace Mantid::API;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+namespace {
+/**
+ * helper method to create a string from min and max extents (with non-zero
+ * signals) ready to be used as the PBins for IntegrateMDHistoWorkspace
+ * algorithm in
+ * exec
+ * @param minVector : Vector containing the minimum extents that we will crop
+ * to.
+ * @param maxVector : Vector containing the maximum extents that we will crop
+ * to.
+ * @param inputWs : Used in the calculation from centre to bin edges
+ * @return : a string vector of binning parameters for IntegrateMDHistoWorkspace
+ * to take as input.
+*/
+std::vector<std::string>
+createPBinStringVector(std::vector<Mantid::coord_t> minVector,
+                       std::vector<Mantid::coord_t> maxVector,
+                       IMDHistoWorkspace_sptr inputWs) {
+  size_t numDims = inputWs->getNumDims();
+  std::vector<std::string> pBinStrVector;
+  for (size_t iter = 0; iter < numDims; iter++) {
+    if (minVector[iter] >= maxVector[iter]) {
```

Comment:
Can this happen? Is this just what happens if your workspace is all zeros? What would happen if I was using this in a script, and I passed in an empty MDHistoWorkspace (all zeros)? Wouldn't this mean that my script would stop processing. I think if your workspace contains all zero's you just leave it alone.